pandas/util/_decorators.py

```diff
@@ -347,3 +347,34 @@ def indent(text: Optional[str], indents: int = 1) -> str:
         return ""
     jointext = "".join(["\n"] + ["    "] * indents)
     return jointext.join(text.split("\n"))
+
+
+_deprecation_registry = {}
+
+
+def strict_deprecation(msg, deprecated_in, remove_in):
+    def deprecator(arg):
+        if isinstance(arg, str):
+            key = arg
```

Comment:
this isnt really necessary, was for a non-decorator usage I had in mind, but that is problematic since it might not run at import-time.